accepted/wasmtime-one-dot-oh.md

```diff
@@ -0,0 +1,464 @@
+
+### Long-term Support Releases
+
+Wasmtime will support some releases for an extended period of time relative to
+other releases. These releases will be known as "long term support" releases or
+LTS releases. **Wasmtime will support two active LTS versions at any point in
+time, each supported for 10 months at a time**.
+
+For users who want to use Wasmtime but are not interested in upgrading monthly,
+this will be a suitable alternative where Wasmtime LTS versions will have the
+following properties while the version is considered "supported":
+
+* They will receive security fixes.
+* They will receive bug fixes related to correctness.
+* They will _not_ receive new features.
+
+For example if a new WebAssembly proposal is implemented it will not be
+backported to LTS versions. Additionally if a bug is fixed in a wasm proposal
+that is off-by-default in an LTS version it will also not be backported. Other
+bug fixes (and of course security fixes) will be backported to supported LTS
+versions.
+
+An LTS version is considered "supported" for 10 release cycles, or 40 weeks (~10
+months). Wasmtime will have two LTS versions at any point in time, with a new
+LTS happening every 5 releases (20 weeks, ~5 months). For ease of remembering
+what's an LTS and what isn't, all releases of Wasmtime divisible by 5 will be
+LTS version, with the exception of 1.0 being an LTS version as well. For example
+the LTS versions of Wasmtime will be 1.0, 5.0, 10.0, 15.0, ...
+
+This cadence means that users who do not want to upgrade Wasmtime monthly will
+be expected to upgrade Wasmtime at least every 10 months, likely every 5 months.
+These upgrades are likely to be less "hassle-free" than each individual version
+upgrade since it will accumulate at least 5 releases worth of minor breaking
+changes.
+
+LTS versions will be maintained in separate branches of the Wasmtime repository.
+At this time it's expected that LTS versions will not needed separate branches
+in separate embedding repos and tags will suffice. The branch names for the
+Wasmtime repository will be `esr-latest` and `esr-oldest` for the most recent
+and the second-most-recent LTS version.
+
+## Backports - Security fixes
+
+With an established concept of releases and LTS for Wasmtime this provides a
+framework to discuss how security issues are handled in Wasmtime. **Security
+issues will be applied to the current version and supported LTS versions of
+Wasmtime**.  Security issues will always be released as patch releases. The
+current version of Wasmtime at the time of the issue being made public will be
+patched in addition to the LTS versions at the time.
+
+For example, if Wasmtime is currently at 12.0 then the current LTS versions are
+5.0 and 10.0. If a security issue is identified at this time then the following
+new releases will be made available: 5.0.1, 10.0.1, 12.0.1. No other versions of
+Wasmtime will be patched, for example 11.0 will not be patched as it's neither
```

Comment:
I wonder if we should include some provision for still updating the previous version if a patch becomes necessary early in the cycle? Otherwise we all of a sudden force people to update within, say, a few days instead of 4 weeks.

This might even point towards regularly doing patch releases for the latest and the previous versions, though that might be more work than we're willing to commit to.

Reply:
Yeah I'll add some words that these are the minimum versions that we patch, but we can always do more.